src/npc.cpp

```diff
@@ -1119,10 +1144,15 @@ void npc::make_angry()
         return; // We're already angry!
     }
 
+    // player allies that become angry should stop being player allies
+    if( is_player_ally() ) {
+        mutiny();
+    }
+
     // Make associated faction, if any, angry at the player too.
-    if( my_fac != nullptr && my_fac->id != faction_id( "no_faction" ) ) {
-        my_fac->likes_u = std::max( -50, my_fac->likes_u - 50 );
-        my_fac->respects_u = std::max( -50, my_fac->respects_u - 50 );
+    if( my_fac && my_fac->id != faction_id( "no_faction" ) && my_fac->id != faction_id( "amf" ) ) {
```

Comment:
This condition indicates that `my_fac` can be null sometimes, but the code in `npc::mutiny` does not check for it. It probably should.

Reply:
Good catch, thanks.